app/src/main/java/com/x8bit/bitwarden/data/platform/manager/model/SpecialCircumstance.kt

```diff
@@ -97,4 +89,23 @@ sealed class SpecialCircumstance : Parcelable {
      */
     @Parcelize
     data object VaultShortcut : SpecialCircumstance()
+
+    /**
+     * A subset of [SpecialCircumstance] that are only relevant in a pre-login state and should be
+     * cleared after a successful login.
+     *
+     * @see [SpecialCircumstanceManager.clearSpecialCircumstanceAfterLogin]
+     */
+    @Parcelize
+    sealed class PreLogin : SpecialCircumstance() {
+        /**
+         * The app was launched via AppLink in order to allow the user complete an ongoing
+         * registration.
+         */
+        @Parcelize
+        data class CompleteRegistration(
```

Comment:
Left this question on Slack but will also put it here for future reference: 

Can you explain the necessity of moving `CompleteRegistration` into the `PreLogin` subclass? Is it just for clarity? I am confused why `CompleteRegistration` wouldn't just be renamed to reflect the pre-login condition since it seems like it's only ever used for that, and there are no other permutations of `PreLogin` (*yet?)

Reply:
Also, none of the others are contained within subclasses so all else being equal it feels like we'd prefer it to be at that same top level.

Reply:
Totally valid question see the [thread](https://github.com/bitwarden/android/pull/3831#discussion_r1731716428) above, also this is the code where I am checking for that status, https://github.com/bitwarden/android/pull/3831/files#diff-4bf84d954e5067e6026bf040a2c54f6c805d1dca3fa36da8f721168027b6e1aaR27-R32

There will be at least on other such case that will be like this. Besides the issue with parcellization and changing the signature of the existing `data object` events I opted this way as it only changes the relevant objects. We could just go with a like an `interface PreLoginEvent` and slap that on any that are applicable but I was kind of following the pattern of the `Internal` sub-sealed classes in some of the VM.
 
Happy to chat on slack/huddle to give more explanation. Also open to other ideas!

Reply:
Well if there is definitely going to be at least one other case that is enough for me. Saw that thread but didn't realize its exact implications so that sounds good!